frontend/src/TooljetDatabase/Table/Footer.jsx

```diff
@@ -70,11 +79,42 @@ const Footer = ({ darkMode, openCreateRowDrawer, totalRecords, fetchTableData })
     fetchTableData(`?limit=${limit}&offset=${offset}`, limit, pageCount - 1);
   };
 
+  const reset = () => {
+    setPageCount(1);
+    setSelectedOption('50 per page');
+    setPageSize(RecordEnum['50 per page']);
+  };
+
+  React.useEffect(() => {
+    reset();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [totalRecords]);
+
+  React.useEffect(() => {
+    if (Object.keys(filters).length === 0 && isMounted) {
+      reset();
+      fetchTableData(`?limit=${pageSize}&offset=0`, pageSize, 1);
+    }
+
+    if (Object.keys(filters).length > 0) {
+      Object.keys(filters).map((key) => {
+        if (!isEmpty(filters[key])) {
+          const { column, operator, value } = filters[key];
+          if (!isEmpty(column) && !isEmpty(operator) && !isEmpty(value)) {
+            fetchTableData(`?limit=${pageSize}&offset=0`, pageSize, 1, filters);
```

Comment:
`limit` and `offset` need to be set in the constructor of `postgrestQueryBuilder`